python/ray/data/_internal/planner/sort.py

```diff
@@ -53,10 +53,9 @@ def fn(
             )
         else:
             boundaries = [(b,) for b in sort_key.boundaries]
+            if sort_key.get_descending()[0]:
+                boundaries = boundaries[::-1]
```

Comment:
does this actually work? what if there are multiple indices for the boundaries, and the first index is the same throughout, and the boundaries are sorted by the second index?

Reply:
This if-statement is only reached if the user provided `boundaries`, and we currently restrict user-passed `boundaries` to apply to the primary sort key (so a single boundary element of `boundaries` a "singleton" tuple with only one index)

Reply:
resolved offline: not an issue

TODO for this if-statement: 
- add documentation